test/Chainweb/Test/Rosetta/RestAPI.hs

```diff
@@ -0,0 +1,513 @@
+        "allowable errors must coincide with failure list"
+
+      step "Check that response statuses are a subset of valid statuses"
+      (_allow_operationStatuses allow `subset` operationStatuses) @?
+        "allowed operation statuses coincide"
+
+      step "Check that response op types are a subset of op types"
+      (_allow_operationTypes allow `subset` allowedOperations) @?
+        "allowed operations coincide"
+  where
+    req0 = NetworkReq nid Nothing
+    respErrors = _allow_errors . _networkOptionsResp_allow
+    allowedOperations = _operationStatus_status <$> operationStatuses
+
+-- | Rosetta network status tests
+--
+networkStatusTests :: RosettaTest
+networkStatusTests tio envIo =
+    testCaseSchSteps "Network Status Tests" $ \step -> do
+      cenv <- envIo
+
+      step "send network status request"
+      resp0 <- networkStatus cenv req
+
+      step "check status response against genesis"
+      genesisId @=? _networkStatusResp_genesisBlockId resp0
+
+      step "send in a transaction and update current block"
+      transferOneAsync_ tio cenv (void . return)
+      resp1 <- networkStatus cenv req
+
+      step "check status response genesis and block height"
+      genesisId @=? _networkStatusResp_genesisBlockId resp1
+      (blockIdOf resp1 > blockIdOf resp0) @? "current block id heights must increment"
+  where
+    req = NetworkReq nid Nothing
+
+    blockIdOf = _blockId_index . _networkStatusResp_currentBlockId
+
+-- ------------------------------------------------------------------ --
+-- Test Data
+
+kda :: Currency
+kda = Currency "KDA" 12 Nothing
+
+nid :: NetworkId
+nid = NetworkId
+    { _networkId_blockchain = "kadena"
+    , _networkId_network = "fastTimedCPM-peterson"
+    , _networkId_subNetworkId = Just (SubNetworkId "0" Nothing)
+    }
+
+genesisId :: BlockId
+genesisId = BlockId 0 "rdfJIktp_WL0oMr8Wr6lH49YkERAJ9MlFp0RPLMXPDE"
+
+rosettaVersion :: RosettaNodeVersion
+rosettaVersion = RosettaNodeVersion
+    { _version_rosettaVersion = "1.3.1"
+    , _version_nodeVersion = "2.0"
+    , _version_middlewareVersion = Nothing
+    , _version_metadata = Just $ HM.fromList
+      [ "node-api-version" A..= ("0.0" :: Text)
+      , "chainweb-version" A..= ("fastTimedCPM-peterson" :: Text)
+      ]
+    }
+
+rosettaFailures :: [RosettaError]
+rosettaFailures = rosettaError <$> enumFrom RosettaChainUnspecified
+
+operationStatuses :: [OperationStatus]
+operationStatuses =
+    [ OperationStatus "Successful" True
+    , OperationStatus "Remediation" True
+    , OperationStatus "CoinbaseReward" True
+    , OperationStatus "FundTx" True
+    , OperationStatus "GasPayment" True
+    , OperationStatus "TransferOrCreateAcct" True
+    ]
+
+-- | Validate all useful data for a tx operation
+--
+validateOp
+    :: Word64
+      -- ^ tx idx
+    -> Text
+      -- ^ operation type
+    -> Text
+      -- ^ operation account name
+    -> Operation
+      -- ^ the op
+    -> Assertion
+validateOp idx opType acct o = do
```

Comment:
Actually, I think this might be more complicated than I initially thought. So I made an issue for later: https://github.com/kadena-io/chainweb-node/issues/1113. Plus, the rosetta validator exercises balances a lot.